### ISSUE TITLE
[a11y]: RadioButtonGroup <fieldset> element does not have a <legend>

### ISSUE THREAD

**Issue by curtispd** [carbon-design-system/carbon#12693]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

10.58.4

### React version

16.14.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker, Latest Deployment, IBM Accessibility

### Description

Accessibility findings
<img width="1728" alt="image" src="https://user-images.githubusercontent.com/2791009/202530345-b369b352-a961-4800-b708-4b0754d4a55e.png">

<fieldset> elements must have a single, non-empty <legend> as a label
     <fieldset> element does not have a <legend>

### WCAG 2.1 Violation

1.3.1 Info and Relationships

### Reproduction/example

https://carbondesignsystem.com/components/radio-button/usage/

### Steps to reproduce

1. Open https://carbondesignsystem.com/components/radio-button/usage/
2. Run accessibility scan

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
